### PR TITLE
refactor: Agent refactoring

### PR DESCRIPTION
**Pull request by maciaszczykm**

### CODE REVIEW

**Comment by maciaszczykm** [comment]
Added restore controller here as well.

**Review by rauerhans — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -16,7 +16,7 @@ $(LOCALBIN):
 
 ENVTEST_K8S_VERSION := 1.28.3
```
Comment (michaeljguarino) [inline Makefile:17]: we should add a make target to copy in the velero crds to simplify that going forward

**Review by michaeljguarino — COMMENTED** [review]
The one thing i'm not sure is duplicated fully is the websocket reconnects, I see the rejoin logic there, how has it been tested?

**Review by michaeljguarino — APPROVED** [review]
add makefile change and should be good to go

**Pull request merged**